### PR TITLE
Nix improvements

### PR DESCRIPTION
**Pull request by sheepforce**
This PR adds pkg-config support for Pysisyphus. It is not actually required or too helpful for python packages, but helps Cabal (the Haskell build tool) find packages as dependencies.

Furthermore, the nix channels has been updated to the just released version of `nixos-20.09`.

### CODE REVIEW

**Comment by sheepforce** [comment]
The build of Pysis is now heavily simplified by using binary caching for all open source dependencies. This reduces both the size of the nix store as well as build time to a few seconds. The docs have been updated accordingly

**Comment by sheepforce** [comment]
The build of Pysis is now heavily simplified by using binary caching for all open source dependencies. This reduces both the size of the nix store as well as build time to a few seconds. The docs have been updated accordingly

File: docs/nix.rst [hunk 1]
```diff
@@ -17,6 +17,25 @@ Please follow the installation instructions in the `nix manual`_ and additional
...
+- MVAPICH2 is used as the commond MPI provider
+- The network for MPI parallelisation is ethernet and the loopback interface
+
+The caching is provided by the Cachix_ service and needs to be enabled for your nix installation:
+
+.. code-block: bash
+    nix-env -iA nixos.cachix
+    cachix use chemix
+
+Cachix will (necessarily) add a new :code:`substituters` and :code:`trusted-public-keys` to your :code:`$HOME/.config/nix/nix.conf` if executed as normal user or :code:`/etc/nix/nix.conf` if executed as root. If written to your user configuration, you need to be a configured as :code:`trusted-users` for nix in :code:`/etc/nix/nix.conf` or your new caching settings will be ignored.
```
Comment (eljost) [inline docs/nix.rst:37]: you need to be [a] configured as ... superfluous "a"

File: nix/pysisyphus.nix [hunk 2]
```diff
@@ -83,10 +83,35 @@ let
         inherit text;
         name = "pysisrc";
       };
+
+  pname = "pysisyphus";
+  version = "dev";
+  description = "Python suite for optimization of stationary points on ground- and excited states PES and determination of reaction paths";
+  homepage = "https://github.com/eljost/pysisyphus";
+
+  pkgConfig = writeTextFile {
```
Comment (eljost) [inline nix/pysisyphus.nix:92]: Now pysisyphus also depends on "scikit-learn". It seems to be available from Nix packages.

File: docs/nix.rst (hunk 1 above)
Comment (sheepforce) [reply docs/nix.rst]: Changed and added more fitting wording for configuration according to the new `config.nix` in  970cf2d.

File: nix/pysisyphus.nix (hunk 2 above)
Comment (sheepforce) [reply nix/pysisyphus.nix:93]: Done in 16d90b4 .

**Pull request merged**